### PR TITLE
Make SignatureContext common in FluffMatchService

### LINKED ISSUE #55: Consider making a common context for constraints from both signatures
Now that every `Type` in `inkuire-engine` should have a DRI, we can consider refactoring `FluffMatchService` to create a common context with constraints from both checked signatures.
Good starting place is marked in code with `//TODO #55`

### PR DESCRIPTION
**Pull request by KacperFKorban**
Closes #55

### CODE REVIEW

File: inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/model/SignatureContext.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.virtuslab.inkuire.engine.model
+
+import cats.kernel.Monoid
+
+case class SignatureContext(
+  vars:        Set[String],
+  constraints: Map[String, Seq[Type]]
+) {
+  // TODO: Issue #28
```
Comment (pikinier20) [inline inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/model/SignatureContext.scala:9]: This issue was closed month ago

File: inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/service/FluffMatchService.scala [hunk 2]
```diff
@@ -19,7 +18,11 @@ class FluffMatchService(val inkuireDb: InkuireDb) extends BaseMatchService with
     inkuireDb.functions.filter { eSgn =>
       signatures.exists { sgn =>
         val ok = for {
-          okTypes <- checkTypes(eSgn.signature, sgn)
+          okTypes <- ancestryGraph.checkVariances(
+            eSgn.signature.types,
+            sgn.types,
+            eSgn.signature.context |+| sgn.context
+          )
```
Comment (pikinier20) [inline inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/service/FluffMatchService.scala:25]: It's kind of unintuitive when we have `okTypes` , then method is named `checkVariances` and then it gets `...types` as arguments, I think we should preserve one convention

File: inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/model/SignatureContext.scala (hunk 1 above)
Comment (KacperFKorban) [reply inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/model/SignatureContext.scala:9]: removed

File: inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/service/FluffMatchService.scala (hunk 2 above)
Comment (KacperFKorban) [reply inkuire-engine/src/main/scala/org/virtuslab/inkuire/engine/service/FluffMatchService.scala:25]: Ok, added `WithVariances` suffixes to accessor anch validation method. It should be more consistent now

**Review by pikinier20 — APPROVED** [review]

**Pull request merged**